lib/puppet/ssl/state_machine.rb

```diff
@@ -260,10 +263,16 @@ def ensure_client_certificate
   private
 
   def run_machine(state, stop)
-    loop do
-      state = state.next_state
+    lock do
+      loop do
+        state = state.next_state
 
-      return state if state.is_a?(stop)
+        break if state.is_a?(stop)
```

Comment:
What does changing this return to a break fix here?  Does the lock block do any kind of retry?  Or is this for graceful/proper exit of the lock block?

Reply:
https://github.com/puppetlabs/puppet/pull/7507#discussion_r282524874